src/lib.rs

```diff
@@ -2147,20 +2147,28 @@ fn ensure_libclang_is_loaded() {
 fn ensure_libclang_is_loaded() {}
 
 /// Error type for rust-bindgen.
-#[derive(Debug)]
+#[derive(Debug, Clone, PartialEq, Eq, Hash)]
 #[non_exhaustive]
 pub enum BindgenError {
-    /// Any provided header was invalid.
-    InvalidHeader(String),
+    /// The header was a folder.
+    FolderAsHeader(String),
+    /// Permissions to read the header is insufficient.
+    InsufficientPermissions(String),
+    /// The header does not exist.
+    NotExist(String),
```

Comment:
Perhaps these should use `PathBuf` instead of `String`?

Reply:
This error is made from `input_header` of `BindgenOptions`, which is `Option<String>`. I will add a process of the conversion into `PathBuf` in `generate` method.